mwdb/core/search/fields.py

```diff
@@ -105,6 +105,18 @@ def ensure_number(value):
     return f'{jsonpath} ? ({" && ".join(conditions)})'
 
 
+def add_escaping_for_like_statement(statement: str) -> str:
+    """
+    Formats query value to work properly with LIKE statements
+    LIKE statements use escaping (default symbol is '\')
+    Every escape should be doubled except '%' and '_'
+    """
+    statement = statement.replace("\\", "\\\\")
+    statement = statement.replace("\\\\%", "\\%")
+    statement = statement.replace("\\\\_", "\\_")
```

Comment:
This can be optimized by negative lookahead
```suggestion
    statement = re.sub(r"\\(?![%_])", r"\\\\", statement)
```

Reply:
Yes, it would be better